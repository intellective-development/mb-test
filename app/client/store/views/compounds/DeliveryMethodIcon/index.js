// @flow

import * as React from 'react';
import _ from 'lodash';
import bindClassNames from 'shared/utils/bind_classnames';
import type { DeliveryMethodType } from 'store/business/delivery_method';
import styles from './index.scss';

const cn = bindClassNames(styles);

type DeliveryMethodIconProps = {delivery_method_type: DeliveryMethodType, width: number, height: number, active?: boolean, className?: string}
const DeliveryMethodIcon = ({delivery_method_type, width, height, active, className}: DeliveryMethodIconProps) => {
  const icon_name = active ? `${delivery_method_type}__red` : delivery_method_type;

  return (
    <img
      alt={delivery_method_type}
      className={cn('cmDeliveryMethodIcon', className)}
      height={height}
      width={width}
      src={`/assets/components/compounds/delivery_method_icon/${icon_name}.svg`} />
  );
};

type MultiDeliveryMethodIconProps = {delivery_method_types: Array<DeliveryMethodType>, width: number, height: number, className?: string};
const MultiDeliveryMethodIcon = ({delivery_method_types, width, height, className}: MultiDeliveryMethodIconProps) => {
  // we uniq these and sort them alphabetically, as the ordering in the prop shouldn't matter
  const uniq_types = _.uniq(delivery_method_types);
  const icon_name = _.sortBy(uniq_types).join('__');

  return (
    <img
      alt={icon_name}
      className={cn('cmDeliveryMethodIcon_Multi', className)}
      height={height}
      width={width}
      src={`/assets/components/compounds/delivery_method_icon/${icon_name}.svg`} />
  );
};

export default DeliveryMethodIcon;
export { MultiDeliveryMethodIcon };
