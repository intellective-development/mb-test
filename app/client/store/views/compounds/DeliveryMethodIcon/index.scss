.cmDeliveryMethodIcon,
.cmDeliveryMethodIcon_Multi {
  margin: 0 12px;
}
