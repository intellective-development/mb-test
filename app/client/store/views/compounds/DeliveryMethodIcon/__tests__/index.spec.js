
import React from 'react';

import DeliveryMethodIcon, { MultiDeliveryMethodIcon } from '../index';

describe('DeliveryMethodIcon', () => {
  const types = [
    ['on_demand'],
    ['pickup'],
    ['shipped']
  ];

  types.forEach(delivery_method_type => {
    it(`renders an icon for ${delivery_method_type}`, () => {
      expect(render(
        <DeliveryMethodIcon delivery_method_type={delivery_method_type} />
      )).toMatchSnapshot();
    });
  });
});

describe('MultiDeliveryMethodIcon', () => {
  const type_permutations = [
    ['on_demand'],
    ['pickup'],
    ['shipped'],
    ['on_demand', 'pickup'],
    ['on_demand', 'shipped'],
    ['pickup', 'shipped'],
    ['shipped', 'pickup'],
    ['shipped', 'pickup'],
    ['on_demand', 'on_demand'],
    ['on_demand', 'shipped', 'on_demand']
  ];

  type_permutations.forEach(delivery_method_types => {
    it(`renders an icon for ${delivery_method_types.join(', ')}`, () => {
      expect(render(
        <MultiDeliveryMethodIcon delivery_method_types={delivery_method_types} />
      )).toMatchSnapshot();
    });
  });
});
