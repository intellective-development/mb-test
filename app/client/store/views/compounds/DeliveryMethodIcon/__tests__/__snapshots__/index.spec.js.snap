// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryMethodIcon renders an icon for on_demand 1`] = `
<img
  alt="on_demand"
  class="cmDeliveryMethodIcon"
  src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
/>
`;

exports[`DeliveryMethodIcon renders an icon for pickup 1`] = `
<img
  alt="pickup"
  class="cmDeliveryMethodIcon"
  src="/assets/components/compounds/delivery_method_icon/pickup.svg"
/>
`;

exports[`DeliveryMethodIcon renders an icon for shipped 1`] = `
<img
  alt="shipped"
  class="cmDeliveryMethodIcon"
  src="/assets/components/compounds/delivery_method_icon/shipped.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for on_demand 1`] = `
<img
  alt="on_demand"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for on_demand, on_demand 1`] = `
<img
  alt="on_demand"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for on_demand, pickup 1`] = `
<img
  alt="on_demand__pickup"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/on_demand__pickup.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for on_demand, shipped 1`] = `
<img
  alt="on_demand__shipped"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/on_demand__shipped.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for on_demand, shipped, on_demand 1`] = `
<img
  alt="on_demand__shipped"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/on_demand__shipped.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for pickup 1`] = `
<img
  alt="pickup"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/pickup.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for pickup, shipped 1`] = `
<img
  alt="pickup__shipped"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/pickup__shipped.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for shipped 1`] = `
<img
  alt="shipped"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/shipped.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for shipped, pickup 1`] = `
<img
  alt="pickup__shipped"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/pickup__shipped.svg"
/>
`;

exports[`MultiDeliveryMethodIcon renders an icon for shipped, pickup 2`] = `
<img
  alt="pickup__shipped"
  class="cmDeliveryMethodIcon_Multi"
  src="/assets/components/compounds/delivery_method_icon/pickup__shipped.svg"
/>
`;
